components/places/src/ffi.rs

```diff
@@ -6,9 +6,105 @@
+    ErrorCode, ExternError, Handle, HandleError,
 };
+use std::sync::Arc;
+
+lazy_static::lazy_static! {
+    pub static ref APIS: ConcurrentHandleMap<Arc<PlacesApi>> = ConcurrentHandleMap::new();
+    pub static ref CONNECTIONS: ConcurrentHandleMap<PlacesDb> = ConcurrentHandleMap::new();
+}
+
+fn parse_url(url: &str) -> crate::Result<url::Url> {
+    Ok(url::Url::parse(url)?)
+}
+
+fn places_get_latest_history_metadata_for_url(
+    handle: i64,
+    url: String,
+) -> Result<Option<HistoryMetadata>, ErrorWrapper> {
+    CONNECTIONS.get(
+        Handle::from_u64(handle as u64)?,
+        |conn| -> Result<_, ErrorWrapper> {
+            let url = parse_url(url.as_str())?;
+            let metadata = crate::storage::history_metadata::get_latest_for_url(conn, &url)?;
+            Ok(metadata)
+        },
+    )
+}
+
+fn places_get_history_metadata_between(
+    handle: i64,
+    start: i64,
+    end: i64,
+) -> Result<Vec<HistoryMetadata>, ErrorWrapper> {
+    log::debug!("places_get_history_metadata_between");
+    CONNECTIONS.get(
+        Handle::from_u64(handle as u64)?,
+        |conn| -> Result<_, ErrorWrapper> {
+            let between = crate::storage::history_metadata::get_between(conn, start, end)?;
+            Ok(between)
+        },
+    )
+}
+
+fn places_get_history_metadata_since(
+    handle: i64,
+    start: i64,
+) -> Result<Vec<HistoryMetadata>, ErrorWrapper> {
+    log::debug!("places_get_history_metadata_since");
+    CONNECTIONS.get(
+        Handle::from_u64(handle as u64)?,
+        |conn| -> Result<_, ErrorWrapper> {
+            let since = crate::storage::history_metadata::get_since(conn, start)?;
+            Ok(since)
+        },
+    )
+}
+
+fn places_query_history_metadata(
+    handle: i64,
+    query: String,
+    limit: i64,
+) -> Result<Vec<HistoryMetadata>, ErrorWrapper> {
+    log::debug!("places_get_history_metadata_since");
+    CONNECTIONS.get(
+        Handle::from_u64(handle as u64)?,
+        |conn| -> Result<_, ErrorWrapper> {
+            let metadata = crate::storage::history_metadata::query(conn, query.as_str(), limit)?;
+            Ok(metadata)
+        },
+    )
+}
+
+fn places_note_history_metadata_observation(
+    handle: i64,
+    data: HistoryMetadataObservation,
+) -> Result<(), ErrorWrapper> {
+    log::debug!("places_note_history_metadata_observation");
+    CONNECTIONS.get(
+        Handle::from_u64(handle as u64)?,
+        |conn| -> Result<_, ErrorWrapper> {
+            crate::storage::history_metadata::apply_metadata_observation(conn, data)?;
+            Ok(())
+        },
+    )
+}
+
+fn places_metadata_delete_older_than(handle: i64, older_than: i64) -> Result<(), ErrorWrapper> {
+    log::debug!("places_note_history_metadata_observation");
+    CONNECTIONS.get(
+        Handle::from_u64(handle as u64)?,
+        |conn| -> Result<_, ErrorWrapper> {
+            crate::storage::history_metadata::delete_older_than(conn, older_than)?;
+            Ok(())
+        },
+    )
+}
```

Comment:
I kinda expected that these bindings would go away in this PR and there will be some include of them out of generated uniffied rs files, but i guess not?

Reply:
Because this is just a partial conversion, these bindings remain because they need to adapt from a "handle" model still used by the most of the FFI. They will go away once the handle goes away.